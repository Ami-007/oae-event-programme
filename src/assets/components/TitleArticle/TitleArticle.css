.title-article{font-family:ptSans;line-height:150%}.title-article .title-article__text{font-size:1.125rem}