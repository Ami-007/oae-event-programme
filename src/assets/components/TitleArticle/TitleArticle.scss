@use "../../sass/variables" as *;
@use "../../sass/typography" as *;

.title-article {
  font-family:ptSans;
  line-height: 150%;



.title-article__header {
  // font-family:$fontOne;
  
}
.title-article__text {
  font-size: 1.125rem;
}
}