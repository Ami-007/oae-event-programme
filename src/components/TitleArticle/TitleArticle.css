@import"https://fonts.googleapis.com/css2?family=PT+Sans&family=Poppins:wght@400;500;600;700&display=swap";@import"https://fonts.googleapis.com/css2?family=Lato&family=Playfair+Display:wght@400;500;600;700&display=swap";.title-article{line-height:150%}.title-article__header{font-family:"Poppins",sans-serif}.title-article__text{font-size:1.125rem;font-family:"PT Sans",sans-serif}