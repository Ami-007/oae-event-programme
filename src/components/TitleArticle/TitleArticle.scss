@use "../../assets/sass/variables" as *;
@use "../../assets/sass/typography" as *;

.title-article {
  line-height: 150%;
  &__header {
    font-family: $modern-header; 
  }
  &__text {
    font-size: 1.125rem;
    font-family: $modern-content;
  }
}
