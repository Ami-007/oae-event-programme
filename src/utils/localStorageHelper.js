export const getDarkTheme = () => {
  //   if (darkTheme == true) {
  //     localStorage.setItem("--background-color", darkTheme);
  //   } else {
  //     localStorage.setItem("theme", "light");
  //   }
  console.log(localStorage.getItem("darkMode"));

  // convert isDarkMode to a Boolean
  // return that boolean

  return true;
};

export const setDarkTheme = (isDarkMode) => {
  localStorage.setItem("darkMode", isDarkMode.toString());
};
