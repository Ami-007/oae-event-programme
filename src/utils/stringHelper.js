export const getFontType = (fontType) => {
  if (typeof fontType === "string") {
    fontType = parseInt(fontType);
  }
  if (fontType === 1) {
    return "modern";
  } else if (fontType === 2) {
    return "classic";
  } else {
    return "modern";
  }
};

export default getFontType;
