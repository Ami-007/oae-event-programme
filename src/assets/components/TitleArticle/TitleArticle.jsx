import React from "react";
import "./TitleArticle.scss";

const TitleArticle = (props) => {
  const {headerTitle, articleText} = props;

  return <div className="title-article">
    <h2 className="title-article__header">{headerTitle}</h2>
    <p className="title-article__text"> {articleText}</p>

  </div>

}

export default TitleArticle;
