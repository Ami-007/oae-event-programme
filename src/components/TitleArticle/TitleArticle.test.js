import { customRender } from "../../utils/testUtils";
import { screen } from "@testing-library/react";
import TitleArticle from "./TitleArticle";

test("Renders TitleArticle component", () => {
  const { container } = customRender(
    <TitleArticle headerTitle="Our Instruments" articleText="Article text" />
  );
  expect(container).toMatchSnapshot();
});

test("Renders header title", () => {
  customRender(<TitleArticle headerTitle="Our Instruments" />);
  const header = screen.queryByText("Our Instruments");
  expect(header).toBeInTheDocument();
});

test("Renders article text", () => {
  customRender(<TitleArticle articleText="Article text" />);
  const text = screen.queryByText("Article text");
  expect(text).toBeInTheDocument();
});
