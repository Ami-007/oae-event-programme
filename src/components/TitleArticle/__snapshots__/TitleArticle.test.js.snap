// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders TitleArticle component 1`] = `
<div>
  <div
    class="title-article"
  >
    <h2
      class="title-article__header"
    >
      Our Instruments
    </h2>
    <p
      class="title-article__text"
    >
       
      Article text
    </p>
  </div>
</div>
`;
